spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/package.scala

```diff
@@ -1,5 +1,115 @@
+    def addTableReadConf(keyspace: String,
+                         table: String,
+                         conf: ReadConf,
+                         cluster: Option[String]) = ConfCache.addTableReadConf(keyspace, table, conf, cluster)
+
+    /** Add keyspace level read configuration settings. Set cluster to None for a single cluster */
+    def addKeyspaceLevelReadConf(keyspace: String,
+                                 conf: ReadConf,
+                                 cluster: Option[String]) =
+      ConfCache.addKeyspaceLevelReadConf(keyspace, conf, cluster)
+
+    /** Add cluster level read configuration settings */
+    def addClusterLevelReadConf(cluster: String, conf: ReadConf) =
+      ConfCache.addClusterLevelReadConf(cluster, conf)
+
+    /** Get read configuration settings by the order of table level, keyspace level, cluster level, default settings */
+    def getReadConf(keyspace: String,
+                    table: String,
+                    cluster: Option[String]): ReadConf =
+      ConfCache.getReadConf(keyspace, table, cluster, ReadConf.fromSparkConf(sqlContext.sparkContext.getConf))
+
+    /** Add table level write configuration settings. Set cluster to None for a single cluster */
+    def addTableWriteConf(keyspace: String,
+                          table: String,
+                          conf: WriteConf,
+                          cluster: Option[String]) =
+      ConfCache.addTableWriteConf(keyspace, table, conf, cluster)
+
+    /** Add keyspace level write configuration settings. Set cluster to None for a single cluster */
+    def addKeyspaceLevelWriteConf(keyspace: String,
+                                  writeConf: WriteConf,
+                                  cluster: Option[String]) =
+      ConfCache.addKeyspaceLevelWriteConf(keyspace, writeConf, cluster)
+
+    /** Add cluster level write configuration settings */
+    def addClusterLevelWriteConf(cluster: String, conf: WriteConf) =
+      ConfCache.addClusterLevelWriteConf(cluster, conf)
+
+    /** Get write configuration settings by the order of table level, keyspace level, cluster level, default settings */
+    def getWriteConf(keyspace: String,
+                     table: String,
+                     cluster: Option[String]): WriteConf =
+      ConfCache.getWriteConf(keyspace, table, cluster, WriteConf.fromSparkConf(sqlContext.sparkContext.getConf))
+
+    /** Add cluster level write configuration settings */
+    def addClusterLevelCassandraConnConf(cluster: String, conf: CassandraConnectorConf) =
+      ConfCache.addClusterLevelCassandraConnConf(cluster, conf)
+
+    /** Get Cassandra connection configuration settings by the order of cluster level, default settings */
+    def getCassandraConnConf(cluster: Option[String]) =
+      ConfCache.getCassandraConnConf(cluster, CassandraConnectorConf(sqlContext.sparkContext.getConf))
+
+    val schemas = CacheBuilder.newBuilder
```

Comment:
However, creating a schema doesn't follow a query to C* all the time. `Schema` object is created from `Metadata` and `Metadata` doesn't query Cassandra for the schema. It caches schema information anyway and this information is updated asynchronously anyway, regardless of whether the user query metadata or not. Google cache probably involves locks and therefore it can be even less efficient in this case.